### ISSUE TITLE
0.27.1: fields with undefined values present in args of generated types

### ISSUE THREAD

**Issue by shawnjones253** [MichalLytek/typegraphql-prisma#434]
**Describe the Bug**
When using:

- `typegraphql-prisma@0.27.1`

AND

- you don't change the output folder for typegraphql-prisma generated types (you must have transpiled code to reproduce this, you can alternately use `emitTranspiledCode = true` to reproduce)

then `Args()` will have ALL keys present when deserialized, even those which the request didn't specify -- those which weren't specified in the request will be present with `undefined` values

Example:

[link](https://github.com/shawnjones253/typegraphql-prisma/blob/845b62866494626e53e829fb3f4f46e479cada53/examples/2-basic/index.ts#L42)
args will be:
```
FindManyUserArgs {
  where: UserWhereInput {
    AND: undefined,
    OR: undefined,
    NOT: undefined,
    id: undefined,
    email: StringFilter {
      equals: 'bob@prisma.io',
      in: undefined,
      notIn: undefined,
      lt: undefined,
      lte: undefined,
      gt: undefined,
      gte: undefined,
      contains: undefined,
      startsWith: undefined,
      endsWith: undefined,
      not: undefined
    },
    name: undefined,
    posts: undefined
  },
  orderBy: undefined,
  cursor: undefined,
  take: undefined,
  skip: undefined,
  distinct: undefined
}
```

**To Reproduce**
git clone https://github.com/shawnjones253/typegraphql-prisma

```
git checkout issue-434-repro
cd typegraphql-prisma/examples/2-basic
npm install
npm run generate
npm run seed
npm run start
```

open localhost:4000 in web browser

send this gql query:
```
{ 
  usersCustom(
  	where: {
    	email: {equals: "bob@prisma.io"}
  	}
	)
  { id }
}
```

**Expected Behavior**
args SHOULD be:
```
FindManyUserArgs {
  where: UserWhereInput { email: StringFilter { equals: 'bob@prisma.io' } }
}
```

this was the behavior with 0.27.0, you can verify this with
` npm install -D -E typegraphql-prisma@0.27.0`
and rerunning the repro above

**Environment (please complete the following information):**

- OS: macOS 14.2
- Node 20.4.0
- `typegraphql-prisma` version 0.27.1
- Prisma version 5.7.1
- TypeScript version 4.9.4

**Additional Context**
I tried to workaround this by using `emitGeneratedCode = false` (this fixes it in the example repo) -- but in my actual project, I ran into tons of `ReferenceError: Cannot access '....' before initialization` despite using all the tsconfig options from [here](https://prisma.typegraphql.com/docs/basics/installation#typescript-configuration)

**Comment by MichalLytek** [comment]
I don't see anything that could affect that:
![image](https://github.com/MichalLytek/typegraphql-prisma/assets/10618781/9d00d86c-5233-490c-9a68-7f4261f63b26)

What is your `type-graphql` version?

**Comment by shawnjones253** [comment]
@MichalLytek  [here's the package.json](https://github.com/shawnjones253/typegraphql-prisma/blob/main/examples/2-basic/package.json) with all the versions used in my repro

`"type-graphql": "2.0.0-beta.4"`

but i can also repro with `2.0.0-beta.3`

**Comment by shawnjones253** [comment]
@MichalLytek it does seem the test snapshots were updated to expect all these undefined values in [this commit](https://github.com/MichalLytek/typegraphql-prisma/commit/0070c6b8362bb9d198f176105994df969db4f720#diff-624e6d9be25797a7d442618aece13b0deb1be5831f959bc13021b3d3e28aafd8)

**Comment by MichalLytek** [comment]
> but i can also repro with 2.0.0-beta.3 and 2.0.0-beta.2

So this is related to `type-graphql` core itself. Will check it.

**Comment by shawnjones253** [comment]
@MichalLytek https://github.com/MichalLytek/typegraphql-prisma/pull/435

**Comment by MichalLytek** [comment]
Closing as fixed by 4c053cc

**Comment by mmmeff** [comment]
Thank you @MichalLytek ! Much appreciated.
